### PR TITLE
Fix duplicate notifications when switching chains

### LINKED ISSUE #103: Notifications: Weird change network behavior creates duplicate notifications
To reproduce:

- make a successful tx on L1
- switch to L2 (cool no problems thanks #102)
- switch back to L1 -- ❗  duplicate notification ❗

### PR DESCRIPTION
**Pull request by garyghayrat**
**Before:**

https://github.com/ScopeLift/l2-flexible-voting-frontend/assets/61768337/76f47d4f-6545-4779-b676-da12935c9dc7

**After:**

https://github.com/ScopeLift/l2-flexible-voting-frontend/assets/61768337/c393ad05-03d4-4e07-be0e-75a44f1920c5

closes #103

### CODE REVIEW

**Comment by garyghayrat** [comment]
Current dependencies on/for this PR:
* main
  * **PR #146** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/146?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/146?utm_source=stack-comment).

File: hooks/useEasyWrite.ts [hunk 1]
```diff
@@ -55,8 +57,14 @@ export const useEasyWrite = (
...
     // succeed
-    if (hash && transactionData?.status === 'success' && prev.status !== 'success') {
+    if (
+      hash &&
+      transactionData?.status === 'success' &&
+      prev.status !== 'success' &&
+      !prev.isNotificationShownAlready
+    ) {
       console.log('notify success');
+      setIsNotificationShownAlready(true);
```
Comment (alexkeating) [inline hooks/useEasyWrite.ts:67]: This works for the first bridge, but if I try to bridge twice than my success toast doesn't get shown. We have to reset the `isNotificationShownAlready` state when a new toast is fired.

Comment (garyghayrat) [reply hooks/useEasyWrite.ts:69]: Fixed here: [7326eda](https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/146/commits/7326edab7fe672b9a62e1e3e34dd93439db1428b)

Comment (garyghayrat) [reply hooks/useEasyWrite.ts:69]: Do you think the overall approach of this PR is good and can be taken out of Draft?

Comment (garyghayrat) [reply hooks/useEasyWrite.ts]: Good idea, like this? https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/146/files

Comment (alexkeating) [reply hooks/useEasyWrite.ts]: Yes, exactly!

Comment (garyghayrat) [reply hooks/useEasyWrite.ts]: Awesome, thanks! will take out of draft now.

**Review by alexkeating — APPROVED** [review]

**Pull request merged**